src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3313,8 +3313,8 @@ bool DoorLockServer::HandleRemoteLockOperation(chip::app::CommandHandler * comma
 
     EndpointId endpoint       = commandPath.mEndpointId;
     OperationErrorEnum reason = OperationErrorEnum::kUnspecified;
-    Nullable<uint16_t> pinUserIdx; // Will get set to non-null if we find a user for the PIN.
-    Optional<uint16_t> pinCredIdx; // Will get set to a value if the PIN is one we know about.
+    Nullable<uint16_t> pinUserIdx;                                  // Will get set to non-null if we find a user for the PIN.
+    Optional<uint16_t> pinCredIdx{ Optional<uint16_t>::Missing() }; // Will get set to a value if the PIN is one we know about.
```

Comment:
Why this change?  The default constructor initializes with Missing already.

Reply:
Right, however it seems some compilers are missing to see it is already initialized and generate the following warning:

> [709/772] Building CXX object CMakeFiles/app.dir/home/kamilkasperczyk/matter/connectedhomeip/src/app/clusters/door-lock-server/door-lock-server.cpp.obj
> /home/kamilkasperczyk/matter/connectedhomeip/src/app/clusters/door-lock-server/door-lock-server.cpp: In member function 'bool DoorLockServer::HandleRemoteLockOperation(chip::app::CommandHandler*, const chip::app::ConcreteCommandPath&, chip::app::Clusters::DoorLock::LockOperationTypeEnum, RemoteLockOpHandler, const chip::Optional<chip::Span<const unsigned char> >&)':
> /home/kamilkasperczyk/matter/connectedhomeip/src/app/clusters/door-lock-server/door-lock-server.cpp:3432:38: warning: 'pinCredIdx.chip::Optional<short unsigned int>::mValue.chip::Optional<short unsigned int>::Value::mData' may be used uninitialized [-Wmaybe-uninitialized]
>  3432 |         foundCred[0].credentialIndex = pinCredIdx.Value();
>       |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
> /home/kamilkasperczyk/matter/connectedhomeip/src/app/clusters/door-lock-server/door-lock-server.cpp:3317:24: note: 'pinCredIdx.chip::Optional<short unsigned int>::mValue.chip::Optional<short unsigned int>::Value::mData' was declared here
>  3317 |     Optional<uint16_t> pinCredIdx; //{ Optional<uint16_t>::Missing() }; // Will get set to a value if the PIN is one we know about.

Reply:
Compiler seems to not understand that `HasValue()` implies that `Value()` is valid. I wonder if there is a way to tell that, however I would rather have the fix be a `// NOLINT` or equivalent in the line using `Value` instead of this workaround.

Reply:
Yeah, it does not make sense that the compiler "understands" that this copy constructor initializes things correctly but the default constructor does not.  In particular, the copy constructor "initializes" things to garbage bytes... ;)

We already have `#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"` in some places.  Perhaps we need it in a few more (like the copy constructor of Optional or something)?